gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -454,18 +520,51 @@ public FileStore getFileStore(Path path) {
   public boolean equals(@Nullable Object other) {
     return this == other
         || other instanceof CloudStorageFileSystemProvider
-        && Objects.equals(gcsService, ((CloudStorageFileSystemProvider) other).gcsService);
+        && Objects.equals(storage, ((CloudStorageFileSystemProvider) other).storage);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(gcsService);
+    return Objects.hash(storage);
   }
 
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
-        .add("gcsService", gcsService)
+        .add("storage", storage)
         .toString();
   }
+
+
+
+  private void throwAsIOException(StorageException oops) throws IOException {
```

Comment:
If you make this method "return" IOException than you can make caller do something like:
`throw throwAsIOException(...)` without the extra
```
 // unreachable, but the compiler doesn't know.
 return null;
```